.github/workflows/pipeline-deploy-app-non-prod.yaml

```diff
@@ -0,0 +1,230 @@
+      tag_overwrite: "yes"
+
+  quality-check:
+    if: ${{ needs.metadata.outputs.environment == 'dev' }}
+    name: "Quality Check"
+    uses: NHSDigital/uec-dos-management/.github/workflows/quality-checks.yaml@latest
+    needs:
+      [
+        metadata,
+      ]
+    with:
+      environment: ${{ needs.metadata.outputs.environment }}
+      workspace: ${{ needs.metadata.outputs.workspace }}
+      stacks: '["github-runner", "terraform_management", "application", "data_migration", "domain-test"]'
+    secrets: inherit
+
+  unit-tests:
+    if: ${{ needs.metadata.outputs.environment == 'dev' }}
+    name: Unit Test
+    needs:
+      [
+        metadata,
+        quality-check,
+      ]
+    uses: NHSDigital/uec-dos-management/.github/workflows/unit-test.yaml@latest
+    with:
+      env: ${{ needs.metadata.outputs.environment }}
+    secrets: inherit
+
+  build-application-artefacts:
+    if: ${{ needs.metadata.outputs.environment == 'dev' }}
+    name: Build application artefacts and push to storage
+    needs:
+      [
+        metadata,
+      ]
+    uses: NHSDigital/uec-dos-management/.github/workflows/build-application.yaml@latest
+    with:
+      directory: 'application'
+      environment: ${{ needs.metadata.outputs.environment }}
+      workspace: ${{ needs.metadata.outputs.workspace }}
+      services:
+        "[
+          'healthcare-services-data-manager',
+          'locations-data-manager',
+          'organisation-affiliations-data-manager',
+          'organisations-data-manager',
+          'questionnaire-responses-data-manager',
+          'questionnaires-data-manager'
+        ]"
+      commit_hash: ${{ needs.metadata.outputs.commit_hash }}
+      artefact_bucket_name: ${{ needs.metadata.outputs.artefact_bucket_name }}
+    secrets: inherit
+
+  build-data-migration-artefacts:
+    if: ${{ needs.metadata.outputs.environment == 'dev' }}
+    name: Build data migration artefacts and push to storage
+    needs:
+      [
+        metadata,
+      ]
+    uses: NHSDigital/uec-dos-management/.github/workflows/build-application.yaml@latest
+    with:
+      directory: 'data_migration'
+      environment: ${{ needs.metadata.outputs.environment }}
+      workspace: ${{ needs.metadata.outputs.workspace }}
+      services:
+        "[
+          'org_data_load',
+          'locations_data_load',
+          'organisation_affiliations_data_load',
+          'service_data_load',
+          'cm_database_populator',
+          'cm_questionnaire_response_populator'
+        ]"
+      commit_hash: ${{ needs.metadata.outputs.commit_hash }}
+      artefact_bucket_name: ${{ needs.metadata.outputs.artefact_bucket_name }}
+    secrets: inherit
+
+  deploy-infrastructure:
+    if: always() && !cancelled() && (needs.quality-check.result == 'success' || needs.quality-check.result == 'skipped')
+    name: Deploy infrastructure to ${{ needs.metadata.outputs.environment }} Environment
+    needs:
+      [
+        quality-check,
+        metadata,
```

Comment:
Deploying infra should depend on building of artefacts.

Reply:
agreed